src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2463 @@
+            matcher=self.matcher,
+            weight_dict=self.weight_dict,
+            eos_coef=config.no_object_weight,
+        )
+
+        self.post_init()
+
+    def get_loss_dict(self, outputs: Dict[str, Tensor], labels: Dict[str, Tensor]) -> Dict[str, Tensor]:
+        loss_dict: Dict[str, Tensor] = self.criterion(outputs, labels)
+        # weight each loss by `self.weight_dict[<LOSS_NAME>]`
+        weighted_loss_dict: Dict[str, Tensor] = {
+            k: v * self.weight_dict[k] for k, v in loss_dict.items() if k in self.weight_dict
+        }
+        return weighted_loss_dict
+
+    def get_loss(self, loss_dict: Dict[str, Tensor]) -> Tensor:
+        # probably an awkward way to reduce it
+        return torch.tensor(list(loss_dict.values()), dtype=torch.float).sum()
+
+    def get_logits(
+        self,
+        outputs: MaskFormerOutput,
+    ) -> Tuple[Tensor, Tensor, List[str, Tensor]]:
+        pixel_embeddings: Tensor = outputs.pixel_decoder_last_hidden_state
+        # get the auxilary predictions (one for each decoder's layer)
+        auxilary_logits: List[str, Tensor] = []
+        # This code is a little bit cumbersome, an improvement can be to return a list of predictions. If we have auxilary loss then we are going to return more than one element in the list
+        if self.config.use_auxilary_loss:
+            stacked_decoder_outputs: Tensor = torch.stack(outputs.transformer_decoder_hidden_states)
+            classes: Tensor = self.class_predictor(stacked_decoder_outputs)
+            class_queries_logits: Tensor = classes[-1]
+            # get the masks
+            mask_embeddings: Tensor = self.mask_embedder(stacked_decoder_outputs)
+            # sum up over the channels for each embedding
+            binaries_masks: Tensor = torch.einsum("lbqc,   bchw -> lbqhw", mask_embeddings, pixel_embeddings)
+            masks_queries_logits: Tensor = binaries_masks[-1]
+            # go til [:-1] because the last one is always used
+            for aux_binary_masks, aux_classes in zip(binaries_masks[:-1], classes[:-1]):
+                auxilary_logits.append({"masks_queries_logits": aux_binary_masks, "class_queries_logits": aux_classes})
+
+        else:
+            transformer_decoder_hidden_states: Tensor = outputs.transformer_decoder_last_hidden_state
+            classes: Tensor = self.class_predictor(transformer_decoder_hidden_states)
+            class_queries_logits: Tensor = classes
+            # get the masks
+            mask_embeddings: Tensor = self.mask_embedder(transformer_decoder_hidden_states)
+            # sum up over the channels
+            masks_queries_logits: Tensor = torch.einsum("bqc,   bchw -> bqhw", mask_embeddings, pixel_embeddings)
+
+        return class_queries_logits, masks_queries_logits, auxilary_logits
+
+    def forward(
+        self,
+        pixel_values: Tensor,
+        mask_labels: Optional[Tensor] = None,
+        class_labels: Optional[Tensor] = None,
```

Comment:
Can we instead have a single `labels` argument similar to how this is done for `DetrForObjectDetection`? It can be a list of dicts, each dictionary containing at least the following 2 keys: 'class_labels' and 'masks' (the class labels and masks of an image in the batch respectively).

Reply:
Original we had a single `labels` but it was changed due to @sgugger suggestions

Reply:
We should have two labels (and Detr will need to be changed to that API at some point too), similarly to the QA models for instance.